sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoinExec.scala

```diff
@@ -71,8 +89,134 @@ case class ShuffledHashJoinExec(
     val numOutputRows = longMetric("numOutputRows")
     streamedPlan.execute().zipPartitions(buildPlan.execute()) { (streamIter, buildIter) =>
       val hashed = buildHashedRelation(buildIter)
-      join(streamIter, hashed, numOutputRows)
+      joinType match {
+        case FullOuter => fullOuterJoin(streamIter, hashed, numOutputRows)
```

Comment:
Should we have a config to disable this feature ?

Reply:
In this current PR implementation, it seems we can disable it if we set -1 to `spark.sql.autoBroadcastJoinThreshold`.

Reply:
I am not sure if that's a good idea: spark.sql.autoBroadcastJoinThreshold is a very widely used config and I think we should have a separate config to disable just this full outer join optimization, without having to turn of BHJ in itself.

Reply:
Yea, I've already asked it above: https://github.com/apache/spark/pull/29342#discussion_r466168517

Reply:
@agrawaldevesh, @maropu - The feature is essentially disabled by default, as shuffled hash join is disabled by default by [config `spark.sql.join.preferSortMergeJoin=true`](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala#L213). So I don't think we need a separate config to disable this feature. In addition, set -1 to `spark.sql.autoBroadcastJoinThreshold` would [disable BHJ](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala#L278) but not SHJ. [SHJ needs the config to be carefully tuned to be enabled, but not -1](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala#L354). So when user explicitly enables shuffled hash join by (1).setting `spark.sql.join.preferSortMergeJoin`=false, and (2).carefully tuning `spark.sql.autoBroadcastJoinThreshold`, they should probably have a clear mind what they are doing.